### PR TITLE
Fix ING adapter scope issue

### PR DESCRIPTION
**Pull request by berezkin88**

### CODE REVIEW

File: adapters/ing-psd2-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ing/internal/service/IngOauth2Service.java [hunk 1]
```diff
@@ -25,16 +25,28 @@ public IngOauth2Service(Oauth2Api oauth2Api,
...
             .getBody();
 
         parameters.setClientId(getClientId());
+        clearParameters(parameters);
 
         return URI.create(StringUri.withQuery(authorizationUrlResponse.getLocation(), parameters.asMap()));
     }
 
+    private void clearParameters(Oauth2Service.Parameters parameters) {
+        if (parameters.get("limit") != null) {
```
Comment (spetrychenko) [inline adapters/ing-psd2-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ing/internal/service/IngOauth2Service.java]: i suggest to use `StringUtils.isBlank` for checking

File: adapters/ing-psd2-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ing/internal/service/IngOauth2Service.java [hunk 2]
```diff
@@ -25,16 +25,28 @@ public IngOauth2Service(Oauth2Api oauth2Api,
...
+            parameters.remove("limit");
+        }
+
+        if (parameters.get("balanceTypes") != null) {
```
Comment (spetrychenko) [inline adapters/ing-psd2-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ing/internal/service/IngOauth2Service.java]: i suggest to use `StringUtils.isBlank` for checking

File: adapters/ing-psd2-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ing/internal/service/ParametersValidationService.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import java.util.stream.Collectors;
+
+public class ParametersValidationService {
+
+    private static final String SCOPE = Oauth2Service.Parameters.SCOPE;
+    private static final String DEFAULT_SCOPE = "payment-accounts:transactions:view";
+    private static final List<String> availableScopes = Arrays.asList(DEFAULT_SCOPE, "payment-accounts:balances:view");
+
+    public static Oauth2Service.Parameters validateScope(Oauth2Service.Parameters params) {
+        if (params.get(SCOPE) == null) {
```
Comment (spetrychenko) [inline adapters/ing-psd2-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ing/internal/service/ParametersValidationService.java]: i suggest to use` StringUtils.isBlank` for checking

File: adapters/ing-psd2-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ing/internal/service/ParametersValidationService.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    private static boolean hasScopes(Oauth2Service.Parameters params) {
+        List<String> listOfScopes = scopesAsList(params);
+
+        return availableScopes.containsAll(listOfScopes);
+    }
+
+    private static List<String> scopesAsList(Oauth2Service.Parameters params) {
+        return Arrays.asList(params.getScope().split(" "));
```
Comment (spetrychenko) [inline adapters/ing-psd2-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ing/internal/service/ParametersValidationService.java]: replace with `.split("\\s+")`

Comment (berezkin88) [reply adapters/ing-psd2-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ing/internal/service/ParametersValidationService.java]: done

File: adapters/ing-psd2-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ing/internal/service/ParametersValidationService.java (hunk 3 above)
Comment (berezkin88) [reply adapters/ing-psd2-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ing/internal/service/ParametersValidationService.java]: done

File: adapters/ing-psd2-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ing/internal/service/IngOauth2Service.java (hunk 2 above)
Comment (berezkin88) [reply adapters/ing-psd2-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ing/internal/service/IngOauth2Service.java]: done

File: adapters/ing-psd2-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ing/internal/service/IngOauth2Service.java (hunk 1 above)
Comment (berezkin88) [reply adapters/ing-psd2-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ing/internal/service/IngOauth2Service.java]: done

File: adapters/ing-psd2-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ing/internal/service/IngOauth2Service.java [hunk 5]
```diff
@@ -25,16 +26,28 @@ public IngOauth2Service(Oauth2Api oauth2Api,
...
+        AuthorizationURLResponse authorizationUrlResponse = oauth2Api.getAuthorizationUrl(clientAuthentication, parameters.getScope())
             .getBody();
 
         parameters.setClientId(getClientId());
+        clearParameters(parameters);
 
         return URI.create(StringUri.withQuery(authorizationUrlResponse.getLocation(), parameters.asMap()));
     }
 
+    private void clearParameters(Oauth2Service.Parameters parameters) {
```
Comment (n0rdy) [inline adapters/ing-psd2-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ing/internal/service/IngOauth2Service.java]: Could you, please, add a short comment above this method to clarify why we remove these parameters? Just for the future.

File: adapters/ing-psd2-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ing/internal/service/ParametersValidationService.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ParametersValidationService {
+
+    private static final String SCOPE = Oauth2Service.Parameters.SCOPE;
+    private static final String DEFAULT_SCOPE = "payment-accounts:transactions:view";
+    // todo add 'payment-accounts:orders:create' scope when PIS flow will be implemented
+    private static final List<String> availableScopes = Arrays.asList(DEFAULT_SCOPE, "payment-accounts:balances:view");
```
Comment (n0rdy) [inline adapters/ing-psd2-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ing/internal/service/ParametersValidationService.java]: As far as I see, this collection is used to check whether it contains some String value or not. Seems like an ideal case for using HashSet to improve the performance. What do you think?

Comment (berezkin88) [reply adapters/ing-psd2-adapter/src/main/java/de/adorsys/xs2a/adapter/service/ing/internal/service/ParametersValidationService.java]: It's probably gonna be only three elements there so not much performance improves, no problem with replacing with Set though.

**Pull request merged**